docs/source/contributing.rst

```diff
@@ -0,0 +1,220 @@
+
+
+Project Structure
+***************************************
+Below is a basic overview of the project structure:
+
+.. code-block:: none
+
+	├── docs/  
+	├── kordac/  
+	│   ├── html-templates/
+	│   ├── KordacExtension.py
+	│   ├── Kordac.py
+	│   ├── processor-info.json
+	│   ├── processors/
+	│   │   └── errors/
+	│   ├── tests/
+	│   │   └── assets/
+	│   └── utils/
+	├── requirements.txt
+	└── setup.py
+
+The files and folders of interest are:
+
+- ``Kordac()``
+	The convertor object itself. This is what a user will use to create a Kordac converter, and what is used to define a custom processor list, custom html templates and custom Markdown Extensions to use.
+
+- ``KordacResult()``
+    The object returned by ``Kordac()`` containing:
+    	- Converted html string
+    	- Title
+    	- Required files (images, interactives, scratch images, page scripts)
+    	- Heading tree
+    	- Required glossary terms
+
+- ``KordacExtension()``
+    This is the main class of the project, and inherits the ``Extension`` class from Markdown.
+    It loads all of the processor information, loads the template files and clears and populates the attributes to be returned by the ``KordacResult`` object.
+
+- ``Processors/``
+  	There is a different processor for each tag. A processor uses it's corresponding regex loaded from ``processor-info.json`` to find matches in the text, and uses the given arguments in the matched tag to populate and output it's html template.
+
+- ``html-templates/``
+  	The html templates (using Jinja2 template engine) to be populated by processors.
+
+- ``processor-info.json``
+	Every processor is listed in this file, and will at least contain a regex pattern to match it's corresponding tag.
+	Most will also define required and optional parameters, these correspond to arguments in the tag's template.
+
+
+
+Creating a New Processor
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+To create a new processor, a good place to start is the `Extension API`_ page of the Python Markdown docs, and you can also read the `source code`_ itself.
+
+.. _Extension API: https://pythonhosted.org/Markdown/extensions/api.html
+
+.. _source code: https://github.com/waylan/Python-Markdown
+
+
+There are several different kinds of processors in Python Markdown, each serving a slightly different purpose. We recommend reading the API docs to determine which processor best suits your purpose (Kordac currently makes use of ``preprocessor``, ``blockprocessor``, ``inlinepattern``, ``treeprocessor`` and ``postprocessor``).
+
+The order of the processors matters and is defined when each processor is added to the ``OrderedDict`` in ``KordacExtension.py``.
+
+Each processor is independent of every other processor. If you have two processors in the pipeline that may overlap (e.g. codehilite and fencedcode), the second processor must handle whatever the first outputs. Therefore refrain from manipulating the output of a processor for a later processor.
```

Comment:
Well they do not have to be independent but they should try to be as much as possible. Other times dependence is necessary as per the ScratchCompatibilityPreprocessor.

If they cannot be independent then compatibility should occur in the processor that happens last (i.e. the downstream processor). That is output should be consistent based on input, not the other way round (e.g. codehilite and fenced_code).